### PR TITLE
refactor: remove dup code

### PR DESCRIPTION
**Pull request by hyschumi**
remove thread_safe_store dup code
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
remove dup code in "thread_safe_store"
#### Which issue(s) this PR fixes:
none

#### Special notes for your reviewer:
none
#### Does this PR introduce a user-facing change?

```
NONE

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```
```

### CODE REVIEW

**Comment by hyschumi** [comment]
I signed it

File: staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go [hunk 1]
```diff
@@ -71,11 +71,7 @@ type threadSafeMap struct {
 }
 
 func (c *threadSafeMap) Add(key string, obj interface{}) {
-	c.lock.Lock()
-	defer c.lock.Unlock()
-	oldObject := c.items[key]
-	c.items[key] = obj
-	c.updateIndices(oldObject, obj, key)
```
Comment (yliaog) [inline staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go:78]: why is updateIndices call removed?

Comment (hyschumi) [reply staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go:78]: the code of `Add` is same as `Update`, we can call  `Update` directly in `Add`

**Comment by hyschumi** [comment]
Anything else to merge this on master?

**Comment by yliaog** [comment]
another question / comment around line 186.

**Comment by hyschumi** [comment]
> another question / comment around line 186.
186 ? can you give me more detail about whay you mean? @yliaog

File: staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go [hunk 2]
```diff
@@ -178,19 +174,13 @@ func (c *threadSafeMap) Index(indexName string, obj interface{}) ([]interface{},
...
 
-	index := c.indices[indexName]
-
-	set := index[indexedValue]
-	list := make([]interface{}, 0, set.Len())
-	for key := range set {
+	list := make([]interface{}, 0, len(keys))
+	for _, key := range keys {
 		list = append(list, c.items[key])
 	}
```
Comment (MadhavJivrajani) [inline staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go:185]: I'm not sure this is correct:
- In the original implementation a read lock is held while reading values from the index, hence making it thread safe.
- In the change done, the read lock is going to be held *only* while `IndexKeys` is executed, post which operations are not under a lock which most likely will lead to issues.

**Review by MadhavJivrajani — COMMENTED** [review]
/ok-to-test
Thanks for your contribution!

File: staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go (hunk 2 above)
Comment (hyschumi) [reply staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go:185]: you are right, i will revert the change

**Pull request merged**